### ISSUE TITLE
Makefile:922: recipe for target 'lua-recursor4.o' failed

### ISSUE THREAD

**Issue by kometchtech** [PowerDNS/pdns#4013]
**version**
`pdns-recursor-4.0.0-rc1`

**configure**
```
configure: 
configure: Configuration summary
configure: =====================
configure: 
configure: PowerDNS Recursor configured with:  '--sysconfdir=/etc/powerdns' '--enable-systemd' '--with-protobuf'
configure: 
configure: CC: gcc
configure: CXX: g++
configure: LD: /usr/bin/ld
configure: CFLAGS: -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -Wall -g -O2
configure: CPPFLAGS: 
configure: CXXFLAGS:  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -Wall -g -O2 -std=gnu++11
configure: LDFLAGS: -Wl,-z -Wl,relro -Wl,-z -Wl,now 
configure: LIBS: 
configure: 
configure: BOOST_CPPFLAGS: 
configure: 
configure: Lua support: lua5.2
configure: Protobuf support: yes
configure: systemd support: yes
configure: 
```
**issue**
```
  CXX      iputils.o
  CXX      ixfr.o
  CXX      json.o
  CXX      lua-recursor4.o
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.9/README.Bugs> for instructions.
Makefile:922: recipe for target 'lua-recursor4.o' failed
make[2]: *** [lua-recursor4.o] Error 4
make[2]: Leaving directory '/usr/local/src/pdns-recursor-4.0.0-rc1'
Makefile:1046: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/usr/local/src/pdns-recursor-4.0.0-rc1'
Makefile:691: recipe for target 'all' failed
make: *** [all] Error 2
```

**OS**
```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 8.5 (jessie)
Release:        8.5
Codename:       jessie
$ uname -rm
3.4.39-s5p6818 armv7l
```

**Comment by rgacogne** [comment]
Hi,

My guess would be that you don't have enough RAM, perhaps you could try adding some swap?

**Comment by kometchtech** [comment]
Memory environment truly missing is 1GB, the swap is not environment??
Try in 2GB of memory.

**Comment by kometchtech** [comment]
I make in 2GB of memory, could compile without problems.
Memory 1GB is tough?

**Comment by Habbie** [comment]
If Lua is disabled, lua-recursor4.cc behaves like a very small stub that just goes "no Lua enabled" if you try to use it.

As for your last set of errors, it turns out compiling with lua is in fact broken. Updating lua-recursor4.cc to match the prototypes in the .hh should fix that.

**Comment by pieterlexis** [comment]
@rgacogne fixed building without lua in c672b54a805fc84b3f5831c22516c781890007ef.
